File: 1768.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        int word1Length = word1.length();
        int word2Length = word2.length();
        StringBuilder strB = new StringBuilder();
        int i = 0, j = 0;
        while (i < word1Length || j < word2Length) {
            if (i < word1Length) {
                strB.append(word1.charAt(i++));
            }
            if (j < word2Length) {
                strB.append(word2.charAt(j++));
            }
        }
        return strB.toString();
    }
}

File: Python/1768.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        w1, w2 = 0, 0
        result = ""
        while w1 < len(word1) or w2 < len(word2):
            if w1 < len(word1) and w2 < len(word2):
                result += word1[w1] + word2[w2]
                w1 += 1
                w2 += 1
            elif w1 < len(word1):
                result += word1[w1]
                w1 += 1
            else:
                result += word2[w2]
                w2 += 1

        return result
